test/integration/singlecluster/controller/core/workload_controller_test.go

```diff
@@ -544,3 +546,169 @@ var _ = ginkgo.Describe("Workload controller", ginkgo.Ordered, ginkgo.ContinueOn
 		})
 	})
 })
+
+var _ = ginkgo.Describe("Workload controller with resource retention", ginkgo.Ordered, ginkgo.ContinueOnFailure, func() {
+	ginkgo.When("manager is setup with tiny retention period", func() {
+		var (
+			ns              *corev1.Namespace
+			createdWorkload kueue.Workload
+			localQueue      *kueue.LocalQueue
+			clusterQueue    *kueue.ClusterQueue
+			flavor          *kueue.ResourceFlavor
+		)
+
+		ginkgo.BeforeAll(func() {
+			fwk.StartManager(
+				ctx, cfg,
+				managerAndControllerSetup(
+					&config.Configuration{
+						ObjectRetentionPolicies: &config.ObjectRetentionPolicies{
+							Workloads: &config.WorkloadRetentionPolicy{
+								AfterFinished: &metav1.Duration{
+									Duration: util.TinyTimeout,
+								},
+							},
+						},
+					},
+				),
+			)
+
+			gomega.Expect(features.SetEnable(features.ObjectRetentionPolicies, true)).To(gomega.Succeed())
+		})
+
+		ginkgo.AfterAll(func() {
+			fwk.StopManager(ctx)
+		})
+
+		ginkgo.BeforeEach(func() {
+			ns = util.CreateNamespaceFromPrefixWithLog(ctx, k8sClient, "core-workload-")
+			flavor = testing.MakeResourceFlavor(flavorOnDemand).Obj()
+			gomega.Expect(k8sClient.Create(ctx, flavor)).Should(gomega.Succeed())
+			clusterQueue = testing.MakeClusterQueue("cq").
+				ResourceGroup(*testing.MakeFlavorQuotas(flavorOnDemand).
+					Resource(corev1.ResourceCPU, "1").Obj()).
+				Obj()
+			localQueue = testing.MakeLocalQueue("q", ns.Name).ClusterQueue("cq").Obj()
+			gomega.Expect(k8sClient.Create(ctx, localQueue)).To(gomega.Succeed())
+		})
+
+		ginkgo.AfterEach(func() {
+			gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+			util.ExpectObjectToBeDeleted(ctx, k8sClient, clusterQueue, true)
+			util.ExpectObjectToBeDeleted(ctx, k8sClient, flavor, true)
+		})
+
+		ginkgo.It("should delete the workload after retention period elapses", func() {
+			// Create workload
```

Comment:
Instead of adding comments use `ginkgo.By`. It will make the failure logs more readable when an issue happens (same for all new test cases)